frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-topology/utils.ts

```diff
@@ -223,14 +229,87 @@ export const getEdgesFromNodes = (nodes: PipelineMixedNodeModel[]): PipelineEdge
     }),
   ).filter((edgeList) => !!edgeList);
 
+export const getFinallyTaskHeight = (allTasksLength: number, disableBuilder: boolean): number => {
+  const FINALLY_TASK_NODE_HEIGHT = NODE_HEIGHT + FINALLY_NODE_VERTICAL_SPACING;
+  return (
+    NODE_HEIGHT +
+    (allTasksLength - 1) * FINALLY_TASK_NODE_HEIGHT +
+    (!disableBuilder ? NODE_HEIGHT : 0) +
+    FINALLY_NODE_PADDING * 2
+  );
+};
+
+export const getLastRegularTasks = (nodes: PipelineMixedNodeModel[]): string[] => {
+  const regularTasks = nodes;
+  const runAfters: string[] = regularTasks.reduce((acc, regularTask) => {
+    const {
+      data: {
+        task: { runAfter },
+      },
+    } = regularTask;
+    if (runAfter && runAfter.length) {
+      runAfter.forEach((ra) => {
+        if (!acc.includes(ra)) {
+          acc.push(ra);
+        }
+      });
+    }
+    return acc;
+  }, []);
```

Comment:
You could just hold onto them in an object as keys and it'll distinct it for you. 
```suggestion
     acc[runAfter] = true;
     ...
  }, {});
  Object.keys(...)
```

Alternatively Lodash has a `uniq` utility: https://lodash.com/docs/3.10.1#uniq


Line 251's `.forEach` should be a `.filter` if you choose to keep it the current design.

Reply:
Used uniq from lodash here.